### PR TITLE
Desktop: Resolves #9857: Back up to a subdirectory of the home directory by default

### LINKED ISSUE #9857: Backup Plugin: Backup to a subdirectory of the home directory by default
### Operating system

Linux

### Joplin version

2.14

### Current behaviour

Currently, the default Simple Backup plugin creates backups in a subdirectory of the profile directory. This could be problematic:
- These backups may be difficult to discover.
- If the profile directory is deleted, backups are also lost.

**To make backups easier to discover, we might:**
1. By default, create backups in a subdirectory of the user's home directory (e.g. `JoplinBackup`).
2. So that different profiles still have different backups, create subdirectories of `JoplinBackup` for each profile, like we do with `~/.config/joplin-desktop`.
3. Add a `README` or similar document to explain the `JoplinBackup` folder.

### PR DESCRIPTION
**Pull request by personalizedrefrigerator**
# Summary

Updates the built-in Simple Backup plugin to apply the following upstream pull requests:
- https://github.com/JackGruber/joplin-plugin-backup/pull/70
- https://github.com/JackGruber/joplin-plugin-backup/pull/69
- https://github.com/JackGruber/joplin-plugin-backup/pull/68

Resolves #9857.

# Testing

Manual testing:
1. Start an existing installation of Joplin with the default backup plugin installed
2. Verify that the "create subfolder for Joplin profile" setting is unchecked (in Simple Backup's advanced settings)
3. Verify that a README.md file is created in the existing backup directory (if using the new setting, `~/JoplinBackup/default/...` or `~/JoplinBackup/default-dev/...`, otherwise, may be in a subdirectory of the profile directory)
4. Close Joplin
5. Remove the profile directory
6. Start Joplin
7. Create a backup
8. Verify that the backup is created in the home directory
9. Create a new profile
10. Create a backup from that profile
    - A notebooke/note may need to be created first
11. Verify that the backup is created in a **subdirectory** of `JoplinBackup` in the home directory (e.g. `/home/user/JoplinBackup/profile-a5ix52ns`)
12. Restore from `all_notebooks.jex` in the profile's backup directory with file > import JEX

**Note**: The latest commit has only been tested on OpenSUSE (dev mode).

The above tests testing have also been done with commit 2e7b466b99a5fafa2ced57a85e823dd3be6ab84c on:
- Ubuntu 23.10 (dev mode)
- MacOS (dev mode)
- Windows 11 (dev mode)

It has also been verified (commit 2e7b466b99a5fafa2ced57a85e823dd3be6ab84c) that Joplin Portable launches on Windows and Simple Backup is still initially disabled. After enabling it, backups seem to be created in the profile directory.

# Possible to-dos/considerations

- [x] Currently, devmode and production mode backups will use the same `JoplinBackup` directory and may overwrite one another (as both use `default` as the ID of the default profile).
    - Fixed by adding a `-dev` to the end of the backup directory (e.g. `~/JoplinBackup/default-dev`) when Joplin is in dev mode.
-  What happens when a user has both a `snap` version of Joplin and an AppImage version? Do they both back up to the same place?
    - I suspect that the `snap` version will back up to a hidden home directory in `/home/primary/user/joplin-desktop/[[version here]]/`, which will likely be difficult to restore from. Something similar may be true with the Flatpak.

### CODE REVIEW

File: packages/lib/models/Setting.ts [hunk 1]
```diff
@@ -146,6 +146,7 @@ export interface Constants {
 	pluginDataDir: string;
 	cacheDir: string;
 	pluginDir: string;
+	homeDir: string;
```
Comment (personalizedrefrigerator) [inline packages/lib/models/Setting.ts:149]: I'm not sure that `homeDir` is the best name for this setting...

Currently, `homeDir` points to `os.homedir()` only when not using the portable application (or the `--profile` flag).

**Comment by personalizedrefrigerator** [comment]
Converting to a draft for the next several hours &mdash; feedback was given on the upstream pull requests. I plan to make changes requested by that feedback and apply the changes here. **Edit**: Done. Now waiting for additional feedback.

**Comment by personalizedrefrigerator** [comment]
TODO: 
- [x] Fix merge conflicts upstream, wait about 1 day. Notify @ JackGruber.

**Comment by personalizedrefrigerator** [comment]
Upstream pull requests have been merged! I'm in the process of updating and testing this PR! This should be done in the next hour or two.

**Comment by laurent22** [comment]
Great, thanks for the update! And thanks @JackGruber for reviewing and merging the pull requests!

**Pull request merged**